src/compact.rs

```diff
@@ -139,12 +140,11 @@ pub trait FindexCompact<
         // Get the values stored in the reindexed chains.
         let mut reindexed_chain_values = HashMap::with_capacity(chains_to_reindex.len());
         for (kwi, chain) in chains_to_reindex {
-            let mut indexed_values = HashSet::new();
-            let blocks = chain.into_iter().flat_map(|(_, v)| v).collect::<Vec<_>>();
-            for bytes in Block::unpad(&blocks)? {
-                indexed_values.insert(IndexedValue::try_from_bytes(&bytes)?);
-            }
-            reindexed_chain_values.insert(kwi.clone(), indexed_values);
+            let blocks = chain
+                .into_iter()
+                .flat_map(|(_, chain_value)| chain_value.as_blocks().to_vec())
```

Comment:
Too bad we require an allocation here with the `to_vec()`… Can we do without it?

Reply:
I have implemented a solution without, but I am not sure this is really a win. I can't get it to compile without collecting after calling `as_blocks()` since it refers to a temporary variable.

This doesn't compile, though it would be great:
```
let blocks = chain
    .into_iter()
    .flat_map(|(_, chain_value)| chain_value.as_blocks()); // <- cannot return reference to local variable `chain_value` returns a reference to data owned by the current function
reindexed_chain_values.insert(kwi.clone(), IndexedValue::from_blocks(blocks)?);
```

This compiles:
```
let blocks: Vec<Block<BLOCK_LENGTH>> = chain
    .into_iter()
    .flat_map(|(_, chain_value)| chain_value.into_blocks())
    .collect();
reindexed_chain_values.insert(kwi.clone(), IndexedValue::from_blocks(blocks.iter())?);
```